backends/imgui_impl_osx.mm

```diff
@@ -376,6 +376,18 @@ static ImGuiKey ImGui_ImplOSX_KeyCodeToImGuiKey(int key_code)
     }
 }
 
+#ifdef IMGUI_IMPL_METAL_CPP
```

Comment:
Should this be `IMGUI_IMPL_METAL_CPP_EXTENSIONS` and/or should the header file use `IMGUI_IMPL_METAL_CPP` instead?

It's not clear if you meant to use the same macro as the Metal backend's C++ support. If it is meant to be the same: Is this actually OK for developers targeting older versions of macOS? (Older versions which maybe have Metal C++ but not full application C++.)

Also should the term "Metal" be used here at all? The `imgui_impl_osx` backend can be used with OpenGL too.

Reply:
It should be `IMGUI_IMPL_METAL_CPP_EXTENSIONS`.

I originally kept it as `IMGUI_IMPL_METAL_CPP`, but thought for backward compatibility it might be best to not use the same preprocessor for Metal-related and AppKit-related stuff.

I kept it "Metal" because that's how it shows up in Apple's examples, as `metal-cpp-extensions`.

Reply:
Sounds good, thanks for clarifying!